packages/types/src/serializer.ts

```diff
@@ -1,14 +1,26 @@
 import { Decoder, Encoder } from "./util";
+import { Endpoint } from "./http";
+
+/**
+ * Interface for object requires an Endpoint set.
+ */
+export interface EndpointBearer {
+  endpoint: Endpoint;
+}
 
 /**
  * Response deserializer util functions for AWS services
  */
-export interface SerializerUtils {
+export interface SerializerUtils extends EndpointBearer {
   utf8Decoder: Decoder;
   base64Encoder: Encoder;
+  endpoint: Endpoint;
```

Comment:
As `SerializerUtils extends EndpointBearer`, and `EndpointBearer` has `endpoint: Endpoint`, this line is not required
```suggestion
```

TypeScript playground: https://bit.ly/31F4tfy

Reply:
I missed it when playing with it. Good catch

Reply:
Resolved in https://github.com/aws/aws-sdk-js-v3/pull/404/commits/4387c11f42651b3c5222af1a6fc8ae999e7b58db